File: C/15429.c
#include <stdio.h>
#include <stdlib.h>

int findKing(int len, int *arr)
{
    int ans;
    for (int i = 1; i < len - 1; i++){
        if (*(arr + i) != *(arr + i - 1) + 1){
            ans = i + 1;
            break;
        }
    }
    return ans;
}

int main()
{
    int test_case, g, ans;
    int *arr;
    scanf("%d", &test_case);
    for (int i = 0; i < test_case; i++){
        scanf("%d", &g);
        arr = malloc(g * sizeof(int));
        for (int j = 0; j < g; j++){
            scanf("%d", arr + j);
        }
        printf("%d\n", findKing(g, arr));
    }

    return 0;
}

File: Python/15429.py
class Gnome:
	def __init__(self, g, arr):
		self._len = g
		self._arr = arr
		self._ans = None

	def find_king(self):
		for i in range(1, self._len - 1):
			if self._arr[i] != self._arr[i - 1] + 1:
				self._ans = i + 1
				break
		print(self._ans)

if __name__ == "__main__":
	test_case = int(input())
	for j in range(test_case):
		arr = [int(i) for i in input().split()]
		Gnome(arr[0], arr[1:]).find_king()
